qiskit/pulse/builder.py

```diff
@@ -2444,9 +2429,7 @@ def x(qubit: int):
         deprecated in the future in favor of tight integration with a circuit
         builder interface which is under development.
 
-    Examples:
-
-    .. jupyter-execute::
+    Examples::
```

Comment:
There were two reasons, one was the build time/reliability concern with executing the blocks when no output is generated, it was adding overhead to the sphinx process when it wasn't really needed. The second was most (if not all I don't recall since I did it so long ago) were emitting a `FutureWarning` when the pulse builder was transpiling under the covers.

Reply:
I think they're in there as a type of documentation test, to ensure the documentation stays up-to-date with any changes in Terra.  If simple examples like this are emitting warnings, we probably _do_ need to change their calling paths to be sure that they don't.  If it's the `FutureWarning` we're talking about below, maybe it really would be good to add a `transpile` option to suppress the warning.

I am concerned a lot about the docs build time too, but I'm hesitant to mess with these checks when I think this one's turned up a useful papercut in the interface change.

Reply:
I'm not convinced there is real value provided by this extra execution. The unittests should be providing the same coverage and if they're not we should improve the tests. Doctests are a thing, but if we want to use them I'd argue we should do that in a dedicated job which is only executing the code examples for testing and had useful error output that was actionable (as opposed to sphinx error messages). I've not played with the doctest builder in a long time (probably back in sphinx 1.x days) but we could look at using that or some other tool to do this.

But just executing every code example as part of a full sphinx build, even if there is no generated output just increases our pain getting things through ci and increase the maintenance burden (honestly mostly on both of us since most of everybody else seems to not be nearly as well versed in the language of sphinx error messages).

Reply:
I've reverted the change here because we're not emitting the warning anymore, but I think there is a larger discussion around how we test code examples in the docs which is outside the scope of this PR. I've opened a new issue about this here: https://github.com/Qiskit/qiskit-terra/issues/7661